lib/key/forwarding.ts

```diff
@@ -66,13 +67,14 @@ async function getEncryptionKeysForForwarding(forwarderKey: PrivateKey) {
  * Whether the given key can be used as input to `generateForwardingMaterial` to setup forwarding.
  */
 export const doesKeySupportForwarding = (forwarderKey: PrivateKey) => (
-    getEncryptionKeysForForwarding(forwarderKey)
+    forwarderKey.isDecrypted() && getEncryptionKeysForForwarding(forwarderKey)
         .then((keys) => keys.length > 0)
         .catch(() => false)
```

Comment:
This makes the type of the return value into `boolean | Promise<boolean>`, which is probably unintentional. To fix that, you'd have to make the function async. And I would also then just write this as
```suggestion
    try {
      return forwarderKey.isDecrypted() && getEncryptionKeysForForwarding(forwarderKey).length > 0;
    } catch {
      return false;
    }
```

Reply:
`getEncryptionKeysForForwarding` is a promise, and adding awaits makes it more verbose than it needs to be IMO